src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+
+    private static StorageDaoImpl storageDao;
+    private static RegistrationServiceImpl registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        user = new User("validuser", "password123", 18);
+    }
+
+    @Test
+    public void register_ValidUser_Success() {
+        User expected = user;
+        User actual = registrationService.register(expected);
+        assertNotNull(actual);
+        assertEquals(expected.getLogin(), actual.getLogin());
+    }
+
+    @Test
+    public void register_NullLogin_ExceptionThrown() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_ShortLogin_ExceptionThrown() {
+        user.setLogin("short");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_NullPassword_ExceptionThrown() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_ShortPassword_ExceptionThrown() {
```

Comment:
It is not clear how short that password is.
Also make tests for edge cases for all fields